processing/src/main/java/io/druid/query/groupby/RowBasedColumnSelectorFactory.java

```diff
@@ -407,22 +415,40 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
     } else {
       return new ColumnValueSelector()
       {
+        @Override
+        public boolean isNull()
+        {
+          return row.get().getRaw(columnName) == null;
+        }
+
         @Override
         public double getDouble()
         {
-          return row.get().getMetric(columnName).doubleValue();
+          Number metric = row.get().getMetric(columnName);
+          if (NullHandling.sqlCompatible() && metric == null) {
+            throw new IllegalStateException("Cannot return double for Null Value");
```

Comment:
suggestion: rephrase the message and others like it as "Cannot return null value as double", I think that's clearer

Reply:
sure, will do

Reply:
done.